### PR TITLE
Feat/windows autolinking

### LINKED ISSUE #490: Does react-native-async-storage v1.13.2 support react-native-windows 0.63?
Does react-native-async-storage v1.13.2 support react-native-windows 0.63? Because I try to build on react-native-windows 0.63 via Visual Studio and get error: [@RNC/AsyncStorage]: NativeModule: AsyncStorage is null
 
"react-native": "0.63.2",
"react-native-windows": "^0.63.0-0",
 "@react-native-async-storage/async-storage": "^1.13.2"

### PR DESCRIPTION
**Pull request by mateusz1913**
## Summary

Hi, since version `react-native-windows` 0.63 there is autolinking implemented (https://microsoft.github.io/react-native-windows/docs/native-modules-autolinking), but apparently windows implementation of this module is not recognized during autolinking process.
In order to support autolinking for windows, implementation should be changed to support RNW 0.62 and RNW 0.63+ according to (https://github.com/microsoft/react-native-windows/issues/6024#issuecomment-696329144) - react-native-camera and react-native-config modules are examples with windows implementation that supports autolinking.

In this PR, I have changed structure of windows implementation to support detecting project and solution files for autolinking mechanism on RNW 0.63 and also updated installation and manual linking docs.

Closes #490 

## Test Plan

Create react-native project with react-native-windows 0.63 version, install async-storage module without manual linking and run project (you can clone https://github.com/mateusz1913/rnwindowstest repo where I set up everything already)

### CODE REVIEW

**Comment by Krizzu** [comment]
Hey @kaiguo , mind having a look here?

**Comment by kaiguo** [comment]
@chrisglein  @jonthysell

File: windows/ReactNativeAsyncStorage/ReactNativeAsyncStorage.vcxproj
Comment (jonthysell) [inline windows/ReactNativeAsyncStorage/ReactNativeAsyncStorage.vcxproj:21]: I would change `$(MSBuildThisFileDirectory)` to `$(SolutionDir)` as per newer recommendations: https://microsoft.github.io/react-native-windows/docs/native-modules-setup#cwinrt

File: windows/ReactNativeAsyncStorage61/ReactNativeAsyncStorage61.vcxproj
Comment (jonthysell) [inline windows/ReactNativeAsyncStorage61/ReactNativeAsyncStorage61.vcxproj:21]: As with the main project, replace `$(MSBuildThisFileDirectory)` with `$(SolutionDir)` as per https://microsoft.github.io/react-native-windows/docs/0.61/native-modules-setup#1-fixing-relative-microsoftreactnative-project-paths

File: website/docs/Installation.md
Comment (jonthysell) [inline website/docs/Installation.md:38]: RNW >= 0.63

File: windows/ReactNativeAsyncStorage/ReactNativeAsyncStorage.vcxproj
Comment (jonthysell) [inline windows/ReactNativeAsyncStorage/ReactNativeAsyncStorage.vcxproj:21]: The latest guidance is to use `$(SolutionDir)` here instead of `$(MSBuildThisFileDirectory)`, as per https://microsoft.github.io/react-native-windows/docs/native-modules-setup#cwinrt

File: windows/ReactNativeAsyncStorage/ReactNativeAsyncStorage.vcxproj
Comment (jonthysell) [inline windows/ReactNativeAsyncStorage/ReactNativeAsyncStorage.vcxproj:3]: You missed one update from `Microsoft.Windows.CppWinRT.2.0.190730.2` to `Microsoft.Windows.CppWinRT.2.0.200316.3` here:

```diff
-<Import Project="$(SolutionDir)packages\Microsoft.Windows.CppWinRT.2.0.200316.3\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('$(SolutionDir)packages\Microsoft.Windows.CppWinRT.2.0.190730.2\build\native\Microsoft.Windows.CppWinRT.props')" />
+<Import Project="$(SolutionDir)packages\Microsoft.Windows.CppWinRT.2.0.200316.3\build\native\Microsoft.Windows.CppWinRT.props" Condition="Exists('$(SolutionDir)packages\Microsoft.Windows.CppWinRT.2.0.200316.3\build\native\Microsoft.Windows.CppWinRT.props')" />
```

Comment (jonthysell) [inline windows/ReactNativeAsyncStorage/ReactNativeAsyncStorage.vcxproj:3]: You can't just update the Microsoft.Windows.CppWinRT version in the vcxproj, you also need to update the version in the `packages.config` file in this folder for this to work properly.

**Review by jonthysell — CHANGES_REQUESTED** [review]
Seems to build fine with autolinking once you make the changes I've recommended.

**Comment by mateusz1913** [comment]
Thanks for review! I applied requested changes and rebased with commit from master

**Comment by Krizzu** [comment]
@jonthysell Thanks for jumping in! Do you think it's ready to go?

**Review by jonthysell — APPROVED** [review]
Looks good to me.

**Review by Krizzu — APPROVED** [review]
Thanks a lot @mateusz1913 @jonthysell , great to have auto-linking for windows in 👏

**Comment by flaviobvds** [comment]
Hi. Sorry if this is a newbie question. But even after the commit I'm still not able to import the changes into my project when I run `npm install @react-native-async-storage/async-storage`
What should I do to get the latest version including the changes comitted in #494?

**Comment by chrisglein** [comment]
> Hi. Sorry if this is a newbie question. But even after the commit I'm still not able to import the changes into my project when I run `npm install @react-native-async-storage/async-storage`
> What should I do to get the latest version including the changes comitted in #494?

Generally I look at the Releases page (available from the home page of any repo). In this case it's this: https://github.com/react-native-async-storage/async-storage/releases
The latest available release is older than this commit. So you'll either need to wait for the next one (release schedule is up to the repo owner, and varies repo to repo) or manually apply the commit to your copy of the code to unblock yourself in the meantime (one of the advantages of modules shipping as source).

**Pull request merged**
